tests/integration/targets/certificate_complete_chain/tasks/main.yml

```diff
@@ -84,4 +84,48 @@
       - cert2_rootchain_alt.chain[:-1] | join('') == lookup('file', 'cert2-altchain.pem', rstrip=False)
       - cert2_rootchain_alt.root == lookup('file', 'cert2-altroot.pem', rstrip=False)
 
+  - name: Find alternate rootchain for cert 2 when complete chain is already presented to the module
+    certificate_complete_chain:
+      input_chain: '{{ lookup("file", "cert2.pem", rstrip=False) ~ lookup("file", "cert2-altchain.pem", rstrip=False) ~ lookup("file", "cert2-altroot.pem", rstrip=True) }}'
+      root_certificates:
+      - '{{ remote_tmp_dir }}/files/roots.pem'
+    register: cert2_complete_chain
+  - debug: var=cert2_complete_chain
+  - name: Verify rootchain for cert 2
+    assert:
+      that:
+      - cert2_complete_chain.complete_chain | join('') == (lookup('file', 'cert2.pem', rstrip=False) ~ lookup('file', 'cert2-altchain.pem', rstrip=False) ~ lookup('file', 'cert2-altroot.pem', rstrip=False))
+      - cert2_complete_chain.chain == []
+      - cert2_complete_chain.root == lookup('file', 'cert2-altroot.pem', rstrip=False)
```

Comment:
style nit: some of these chained lookups could be slightly cleaner by going multi-line with the templating, and also defining it once in `vars:` instead of duplicated.

Reply:
I tried to improve this in 2a0624d0c609176dac2f3450e46332a4d1d31252.